src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -1468,17 +1472,19 @@ internal sealed class CSharpLazyLocalFunctionOperation : LazyLocalFunctionOperat
     {
         private readonly CSharpOperationFactory _operationFactory;
         private readonly BoundLocalFunctionStatement _localFunctionStatement;
+        private readonly ImmutableArray<ConditionalOperation> _prependedNullChecks;
 
-        internal CSharpLazyLocalFunctionOperation(CSharpOperationFactory operationFactory, BoundLocalFunctionStatement localFunctionStatement, IMethodSymbol symbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+        internal CSharpLazyLocalFunctionOperation(CSharpOperationFactory operationFactory, BoundLocalFunctionStatement localFunctionStatement, IMethodSymbol symbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ImmutableArray<ConditionalOperation> nullChecksToPrepend, Optional<object> constantValue, bool isImplicit) :
             base(symbol, semanticModel, syntax, type, constantValue, isImplicit)
         {
             _operationFactory = operationFactory;
             _localFunctionStatement = localFunctionStatement;
+            _prependedNullChecks = nullChecksToPrepend;
         }
 
         protected override IBlockOperation CreateBody()
         {
-            return (IBlockOperation)_operationFactory.Create(_localFunctionStatement.Body);
+            return CSharpOperationFactory.Helper.CreateNullCheckedIOperationBody(_operationFactory, _localFunctionStatement.Body, _prependedNullChecks, this);
```

Comment:
>CSharpOperationFactory.Helper [](start = 19, length = 29)

Consider making the helper method an instance method on `CSharpOperationFactory`.